From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 03 Apr 2026 13:00:00 +0200
Subject: dts: Add device tree for decktrix custom PCB

The decktrix board is based on STM32MP157F with the same RAM as the
Discovery Kit but uses a different pinout and peripherals:

- Jadard JD9365TN display (different reset/backlight GPIOs)
- Jadard touch controller on I2C1 (different IRQ/reset GPIOs)
- Cypress CYW43438 WiFi/BT (different reset/shutdown GPIOs)
- BT on UART7 instead of USART2 (no RTS/CTS)
- STPMIC1 PMIC on I2C4 at PD12/PD13 instead of PZ4/PZ5

Signed-off-by: Kirill Yatsenko <kiriyatsenko@gmail.com>
---
 arch/arm/boot/dts/st/Makefile          |   1 +
 arch/arm/boot/dts/st/decktrix-v1.dts   | 163 ++++++++++++++++++++++++
 2 files changed, 164 insertions(+)
 create mode 100644 arch/arm/boot/dts/st/decktrix-v1.dts

diff --git a/arch/arm/boot/dts/st/Makefile b/arch/arm/boot/dts/st/Makefile
index 75346d084d2d..8b3d8d4c9e1f 100644
--- a/arch/arm/boot/dts/st/Makefile
+++ b/arch/arm/boot/dts/st/Makefile
@@ -62,6 +62,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-dk2-scmi.dtb \
 	stm32mp157c-dk2-jadard.dtb \
+	decktrix-v1.dtb \
 	stm32mp157c-ed1.dtb \
 	stm32mp157c-ed1-scmi.dtb \
 	stm32mp157c-emsbc-argon.dtb \
diff --git a/arch/arm/boot/dts/st/decktrix-v1.dts b/arch/arm/boot/dts/st/decktrix-v1.dts
new file mode 100644
index 000000000000..000000000000
--- /dev/null
+++ b/arch/arm/boot/dts/st/decktrix-v1.dts
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) Decktrix Lab
+ * Author: Kirill Yatsenko <kiriyatsenko@gmail.com>
+ *
+ * Decktrix v1 custom PCB based on STM32MP157F.
+ * Same SoC and RAM as STM32MP157F-DK2 but with custom pinout:
+ *   - Jadard JD9365TN display (different reset/backlight GPIOs)
+ *   - Jadard touch on I2C1 (different IRQ/reset GPIOs)
+ *   - Cypress CYW43438 WiFi/BT (different control GPIOs)
+ *   - BT on UART7 (no RTS/CTS)
+ *   - STPMIC1 PMIC on I2C4 at PD12/PD13 (not PZ4/PZ5)
+ */
+
+/dts-v1/;
+
+#include "stm32mp157.dtsi"
+#include "stm32mp15xc.dtsi"
+#include "stm32mp15-pinctrl.dtsi"
+#include "stm32mp15xxac-pinctrl.dtsi"
+#include "stm32mp15xx-dkx.dtsi"
+
+/ {
+	model = "Decktrix v1";
+	compatible = "decktrix,v1", "st,stm32mp157";
+
+	aliases {
+		ethernet0 = &ethernet0;
+		serial3 = &uart7;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	wifi_pwrseq: wifi-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		reset-gpios = <&gpioe 6 GPIO_ACTIVE_LOW>;
+	};
+};
+
+&cryp1 {
+	status = "okay";
+};
+
+&dsi {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	/* Jadard JD9365TN 480x1080 MIPI DSI panel */
+	panel@0 {
+		compatible = "shenzen,z34014p30365ty1", "jadard,jd9365tn";
+		reg = <0>;
+		reset-gpios     = <&gpiod 11 GPIO_ACTIVE_HIGH>;
+		backlight-gpios = <&gpiof 6  GPIO_ACTIVE_HIGH>;
+		status = "okay";
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&dsi_out>;
+			};
+		};
+	};
+};
+
+&dsi_in {
+	remote-endpoint = <&ltdc_ep1_out>;
+};
+
+&dsi_out {
+	remote-endpoint = <&panel_in>;
+};
+
+/* PMIC (STPMIC1) on I2C4: SCL=PD12, SDA=PD13 (AF4) */
+&i2c4 {
+	pinctrl-0 = <&i2c4_pins_decktrix>;
+	pinctrl-1 = <&i2c4_sleep_pins_decktrix>;
+};
+
+&i2c1 {
+	jadard@68 {
+		compatible = "jadard,jdcommon";
+		reg = <0x68>;
+		interrupts-extended = <&gpiog 7 IRQ_TYPE_EDGE_FALLING>;
+		jadard,panel-sense-nums = <30 48>;
+		jadard,panel-coords = <0 480 0 1080>;
+		jadard,panel-max-points = <10>;
+		jadard,int-is-edge = <1>;
+		jadard,irq-gpio = <&gpiog 7 GPIO_ACTIVE_HIGH>;
+		jadard,rst-gpio = <&gpioe 10 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+};
+
+&ltdc {
+	status = "okay";
+
+	port {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		ltdc_ep1_out: endpoint@1 {
+			reg = <1>;
+			remote-endpoint = <&dsi_in>;
+		};
+	};
+};
+
+&pinctrl {
+	i2c4_pins_decktrix: i2c4-decktrix {
+		pins {
+			pinmux = <STM32_PINMUX('D', 12, AF4)>, /* I2C4_SCL */
+				 <STM32_PINMUX('D', 13, AF4)>; /* I2C4_SDA */
+			bias-disable;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+
+	/omit-if-no-ref/
+	i2c4_sleep_pins_decktrix: i2c4-sleep-decktrix {
+		pins {
+			pinmux = <STM32_PINMUX('D', 12, ANALOG)>,
+				 <STM32_PINMUX('D', 13, ANALOG)>;
+		};
+	};
+};
+
+&rtc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rtc_rsvd_pins_a>;
+
+	rtc_lsco_pins_a: rtc-lsco-0 {
+		pins = "out2_rmp";
+		function = "lsco";
+	};
+};
+
+/* Wifi */
+&sdmmc2 {
+	pinctrl-names = "default", "opendrain", "sleep";
+	pinctrl-0 = <&sdmmc2_b4_pins_a>;
+	pinctrl-1 = <&sdmmc2_b4_od_pins_a>;
+	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
+	non-removable;
+	cap-sdio-irq;
+	st,neg-edge;
+	bus-width = <4>;
+	vmmc-supply = <&v3v3>;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	brcmf: wifi@1 {
+		reg = <1>;
+		compatible = "brcm,bcm4329-fmac";
+		pinctrl-names = "default";
+		pinctrl-0 = <&rtc_lsco_pins_a>;
+	};
+};
+
+/* Bluetooth - Cypress CYW43438 on UART7 (no RTS/CTS) */
+&uart7 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&uart7_pins_c>;
+	pinctrl-1 = <&uart7_sleep_pins_c>;
+	status = "okay";
+
+	bluetooth {
+		shutdown-gpios = <&gpiob 10 GPIO_ACTIVE_HIGH>;
+		compatible = "brcm,bcm43438-bt";
+		max-speed = <3000000>;
+		vbat-supply = <&v3v3>;
+		vddio-supply = <&v3v3>;
+	};
+};
-- 
2.43.0
